flocker/node/agents/ebs.py

```diff
@@ -144,17 +145,11 @@ def __init__(self, ec2_client, cluster_id):
         self.zone = ec2_client.zone
         self.cluster_id = cluster_id
 
-    def _blockdevicevolume_from_ebs_volume(self, ebs_volume):
+    def storage_backend_id(self):
         """
-        Helper function to convert Volume information from
-        EBS format to Flocker block device format.
+        Look up the EC2 instance ID for this node.
         """
-        return BlockDeviceVolume(
-            blockdevice_id=unicode(ebs_volume.id),
-            size=int(GB(ebs_volume.size).to_Byte().value),
-            host=None,
-            dataset_id=UUID(ebs_volume.tags[DATASET_ID_LABEL])
-        )
+        return get_instance_metadata()['instance-id'].decode("ascii")
```

Comment:
Thanks for adding this! i am making rest of relevant changes in my EBS attach_volume branch.